$NetBSD: patch-aa,v 1.3 2008/09/18 20:31:26 bjs Exp $

This patch ensures the driver won't use the DRM's MODESET_CTL 
unless direct rendering is enabled.

From Owain Ainsworth <oga@openbsd.org>.

--- src/legacy_crtc.c.orig	2008-06-25 02:56:40.000000000 -0400
+++ src/legacy_crtc.c
@@ -625,6 +625,9 @@ radeon_crtc_modeset_ioctl(xf86CrtcPtr cr
     RADEONCrtcPrivatePtr radeon_crtc = crtc->driver_private;
     struct drm_modeset_ctl modeset;
 
+    if (!info->directRenderingEnabled)
+	return;
+
     modeset.crtc = radeon_crtc->crtc_id;
     modeset.cmd = post ? _DRM_POST_MODESET : _DRM_PRE_MODESET;
 
